table/tables/partition.go

```diff
@@ -122,6 +125,8 @@ type PartitionExpr struct {
 	Column      *expression.Column
 	Ranges      []expression.Expression
 	UpperBounds []expression.Expression
+	// OrigExpr is the partition expression ast.
+	OrigExpr ast.ExprNode
```

Comment:
I'm wondering if we store the expression instead of the ast node, maybe the performance could be better? Not sure because session context may change.

Reply:
The current problem is that `PartitionExpr` is too slow, it will travel all partition and rebuild partition expression.

Reply:
I try to save the ast node and rewrite during call `PartitionExpr()`, but it cause race now.

Reply:
Oh, I see